comp/metadata/packagesigning/packagesigningimpl/aptsigning.go

```diff
@@ -0,0 +1,185 @@
+		if files, err := os.ReadDir(sourceList); err == nil {
+			for _, file := range files {
+				reposPerKey := parseSourceListFile(filepath.Join(sourceList, file.Name()))
+				for name, repos := range reposPerKey {
+					decryptGPGFile(allKeys, repoFile{name, repos}, "signed-by", client, logger)
+				}
+			}
+		}
+	}
+}
+
+func parseSourceListFile(filePath string) map[string][]pkgUtils.Repositories {
+	reposPerKey := make(map[string][]pkgUtils.Repositories)
+	file, err := os.Open(filePath)
+	if err != nil {
+		return nil
+	}
+	defer file.Close()
+	scanner := bufio.NewScanner(file)
+	for scanner.Scan() {
+		line := scanner.Text()
+		if strings.HasPrefix(line, "#") {
+			continue
+		}
+		splitLine := sourceListRegexp.FindStringSubmatch(line)
+		if len(splitLine) > 1 {
+			options := splitLine[2]
+			keyURI := signedBy.FindStringSubmatch(options)
+			if len(keyURI) > 1 {
+				if _, ok := reposPerKey[keyURI[1]]; !ok {
+					reposPerKey[keyURI[1]] = []pkgUtils.Repositories{{RepoName: strings.ReplaceAll(splitLine[3], " ", "/")}}
+				} else {
+					reposPerKey[keyURI[1]] = append(reposPerKey[keyURI[1]], pkgUtils.Repositories{RepoName: strings.ReplaceAll(splitLine[3], " ", "/")})
+				}
+			}
+		}
+	}
+	return reposPerKey
+}
+
+func getDebsigKeyPaths() []string {
+	filePaths := make(map[string]struct{})
+	// Search in the policy files
+	if _, err := os.Stat(debsigPolicies); err == nil {
+		if debsigDirs, err := os.ReadDir(debsigPolicies); err == nil {
+			for _, debsigDir := range debsigDirs {
+				if debsigDir.IsDir() {
+					if policyFiles, err := os.ReadDir(filepath.Join(debsigPolicies, debsigDir.Name())); err == nil {
+						for _, policyFile := range policyFiles {
+							// Get the gpg file name from policy files
+							if debsigFile := getDebsigFileFromPolicy(filepath.Join(debsigPolicies, debsigDir.Name(), policyFile.Name())); debsigFile != "" {
+								debsigFilePath := filepath.Join(debsigKeyring, debsigDir.Name(), debsigFile)
+								if _, err := os.Stat(debsigFilePath); err == nil {
+									filePaths[debsigFilePath] = struct{}{}
+								}
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+	// Denormalise the map
+	filePathsSlice := make([]string, 0)
+	for k := range filePaths {
+		filePathsSlice = append(filePathsSlice, k)
+	}
+	return filePathsSlice
+}
+
+// policy structure to unmarshall the policy files
+type policy struct {
+	XMLName      xml.Name `xml:"Policy"`
+	Origin       origin   `xml:"Origin"`
+	Selection    selection
+	Verification verification
+}
+
+type origin struct {
+	Name        string `xml:"Name,attr"`
+	ID          string `xml:"id,attr"`
+	Description string `xml:"Description,attr"`
+}
+
+type selection struct {
+	Required required `xml:"Required"`
+}
+
+type required struct {
+	Type string `xml:"Type,attr"`
+	File string `xml:"File,attr"`
+	ID   string `xml:"id,attr"`
+}
+
+type verification struct {
+	MinOptional int      `xml:"MinOptional,attr"`
+	Required    required `xml:"Required"`
+}
```

Comment:
Since all of this information is private and not exposed to users of this package I'm wondering if is worth using an anonymous struct https://blog.boot.dev/golang/anonymous-structs-golang/#when-should-i-use-an-anonymous-struct inside the `getDebsigFileFromPolicy`

Reply:
Thanks for the suggestion, however I'm not sure it would make the code more readable. As you say the structs are private so the risk if misusage is low, wdyt?

Reply:
> As you say the structs are private so the risk if misusage is low, wdyt?

Yeah, that works as well 😄